src/thunderbots/external/g3log.BUILD

```diff
@@ -0,0 +1,38 @@
+# Description:
+#   G3log is a logging library that allows us to publish at multiple logging levels,
+#   log to custom sinks, etc.
+
+# G3log uses cmake to generate some files, so we need to run it
+# before we build
+genrule(
+    name = "g3log_cmake",
+    outs = ["include/g3log/generated_definitions.hpp"],
+    srcs = glob(["**/*"]),
+    cmd = "\n".join([
+        # Run cmake, silencing both stdout (">") and stderr ("2>")
+        "cmake external/g3log > /dev/null 2> /dev/null",
+        # Copy the generated header to the location bazel expects it
+        "mv include/g3log/generated_definitions.hpp $@",
+    ]),
+)
+
+cc_library(
+    name = "all",
```

Comment:
One this I noticed, maybe we shouldn't use `all` as a name for our libraries as its internally used by Bazel.

Ive been getting these in Clion: WARNING: The target pattern '//software/geom:all' is ambiguous: ':all' is both a wildcard, and the name of an existing cc_library rule; using the latter interpretation

Reply:
Great catch! In all the cases where we were using `:all`, it made sense to use the library name instead. So instead of declaring a `all` library in `geom`, I just declare a library called `geom`, which lets me write dependencies as `//software/geom` instead of `//software/geom:all`